src/ImageSharp/MetaData/ImageMetaData.cs

```diff
@@ -125,10 +132,46 @@ public double VerticalResolution
         public IList<ImageProperty> Properties { get; } = new List<ImageProperty>();
 
         /// <summary>
-        /// Gets or sets the number of times any animation is repeated.
-        /// <remarks>0 means to repeat indefinitely.</remarks>
+        /// Adds or updates the specified key and value to the <see cref="ImageMetaData"/>.
+        /// </summary>
+        /// <param name="key">The key of the metadata to add.</param>
+        /// <param name="value">The value of the element to add.</param>
+        /// <exception cref="ArgumentNullException">key is null.</exception>
+        /// <exception cref="ArgumentNullException">value is null.</exception>
+        /// <exception cref="ArgumentException">An element with the same key already exists in the <see cref="ImageMetaData"/>.</exception>
+        public void AddOrUpdateFormatMetaData(IImageFormat key, IImageFormatMetaData value)
+        {
+            // Don't think this needs to be threadsafe.
+            Guard.NotNull(value, nameof(value));
+            this.metaData[key] = value;
+        }
+
+        /// <summary>
+        /// Gets the metadata value associated with the specified key.
         /// </summary>
-        public ushort RepeatCount { get; set; }
+        /// <typeparam name="T">The type of metadata.</typeparam>
+        /// <param name="key">The key of the value to get.</param>
+        /// <returns>
+        /// The <typeparamref name="T"/>.
+        /// </returns>
+        public T GetOrAddFormatMetaData<T>(IImageFormat key)
+            where T : IImageFormatMetaData, new()
```

Comment:
Can't decide whether I like the `new()` constraint or not. Seems reasonable here, but I've read pretty bad things about it:
https://blogs.msdn.microsoft.com/seteplia/2017/02/01/dissecting-the-new-constraint-in-c-a-perfect-example-of-a-leaky-abstraction/

(The performance concern does not apply here, for sure.)

Reply:
Actually, the constraint can be removed, if we follow the proposal in my gist!

We just need a factory method in `IImageFormat<TFormatMetaData>`:

```C#
public interface IImageFormat<TFormatMetaData> : IImageFormat
{
	TFormatMetaData CreateDefaultMetaData();
}
```

Reply:
I like this!!!